src/runtime/runtime_mimxrt1062_mpu.go

```diff
@@ -0,0 +1,349 @@
+	RBAR    volatile.Register32 // 0x00C (R/W) - MPU Region Base Address Register
+	RASR    volatile.Register32 // 0x010 (R/W) - MPU Region Attribute and Size Register
+	RBAR_A1 volatile.Register32 // 0x014 (R/W) - MPU Alias 1 Region Base Address Register
+	RASR_A1 volatile.Register32 // 0x018 (R/W) - MPU Alias 1 Region Attribute and Size Register
+	RBAR_A2 volatile.Register32 // 0x01C (R/W) - MPU Alias 2 Region Base Address Register
+	RASR_A2 volatile.Register32 // 0x020 (R/W) - MPU Alias 2 Region Attribute and Size Register
+	RBAR_A3 volatile.Register32 // 0x024 (R/W) - MPU Alias 3 Region Base Address Register
+	RASR_A3 volatile.Register32 // 0x028 (R/W) - MPU Alias 3 Region Attribute and Size Register
+}
+
+var MPU = (*MPU_Type)(unsafe.Pointer(uintptr(0xe000ed90)))
+
+func initCache() {
+	MPU.initialize()
+}
+
+func (mpu *MPU_Type) initialize() {
+
+	mpu.enable(false)
+
+	// -------------------------------------------------------- OVERLAY REGIONS --
+
+	// add Default [0] region to deny access to whole address space to workaround
+	// speculative prefetch. Refer to Arm errata 1013783-B for more details.
+
+	// [0] Default {OVERLAY}:
+	//          4 GiB, -access, @device, -exec, -share, -cache, -buffer, -subregion
+	mpu.setRBAR(0, 0x00000000)
+	mpu.setRASR(rs4GB, apNone, exDevice, false, false, false, false, false)
+
+	// [1] Peripherals {OVERLAY}:
+	//          64 MiB, +ACCESS, @device, +EXEC, -share, -cache, -buffer, -subregion
+	mpu.setRBAR(1, 0x40000000)
+	mpu.setRASR(rs64MB, apFull, exDevice, true, false, false, false, false)
+
+	// [2] RAM {OVERLAY}:
+	//          1 GiB, +ACCESS, @device, +EXEC, -share, -cache, -buffer, -subregion
+	mpu.setRBAR(2, 0x00000000)
+	mpu.setRASR(rs1GB, apFull, exDevice, true, false, false, false, false)
+
+	// ----------------------------------------------------- PERIPHERAL REGIONS --
+
+	// [3] ITCM:
+	//          512 KiB, +ACCESS, #NORMAL, +EXEC, -share, -cache, -buffer, -subregion
+	mpu.setRBAR(3, 0x00000000)
+	mpu.setRASR(rs512KB, apFull, exNormal, true, false, false, false, false)
+
+	// [4] DTCM:
+	//          512 KiB, +ACCESS, #NORMAL, +EXEC, -share, -cache, -buffer, -subregion
+	mpu.setRBAR(4, 0x20000000)
+	mpu.setRASR(rs512KB, apFull, exNormal, true, false, false, false, false)
+
+	// [5] RAM (AXI):
+	//          512 KiB, +ACCESS, #NORMAL, +EXEC, -share, +CACHE, +BUFFER, -subregion
+	mpu.setRBAR(5, 0x20200000)
+	mpu.setRASR(rs512KB, apFull, exNormal, true, false, true, true, false)
+
+	// [6] FlexSPI:
+	//          512 MiB, +ACCESS, #NORMAL, +EXEC, -share, +CACHE, +BUFFER, -subregion
+	mpu.setRBAR(6, 0x70000000)
+	mpu.setRASR(rs512MB, apFull, exNormal, true, false, true, true, false)
+
+	// [7] QSPI flash:
+	//          2 MiB, +ACCESS, #NORMAL, +EXEC, -share, +CACHE, +BUFFER, -subregion
+	mpu.setRBAR(7, 0x60000000)
+	mpu.setRASR(rs2MB, apFull, exNormal, true, false, true, true, false)
+
+	mpu.enable(true)
+}
+
+// MPU Type Register Definitions
```

Comment:
Seems like `device` bitfield stuff too.

Reply:
Moved to new file `src/device/nxp/mpu_mimxrt1062.go`